File: DecompiledGameScripts/core/gameplay/prereqs/damage/attackTypeHitPrereqCondition.swift

public class AttackTypeHitPrereqCondition extends BaseHitPrereqCondition {

  public let m_attackType: gamedataAttackType;

  public func SetData(recordID: TweakDBID) -> Void {
    let str: String = TweakDBInterface.GetString(recordID + t".attackType", "");
    let result: gamedataAttackType = IntEnum<gamedataAttackType>(Cast<Int32>(EnumValueFromString("gamedataAttackType", str)));
    if EnumInt(result) < 0 {
      this.m_attackType = gamedataAttackType.Invalid;
    } else {
      this.m_attackType = result;
    };
    super.SetData(recordID);
  }

  public func Evaluate(hitEvent: ref<gameHitEvent>) -> Bool {
    let result: Bool = Equals(hitEvent.attackData.GetAttackType(), this.m_attackType);
    if result {
      result = this.CheckOnlyOncePerShot(hitEvent);
    };
    return this.m_invert ? !result : result;
  }
}

File: Definitions/Game_Definitions/classes/AttackTypeHitPrereqCondition.lua
---@meta
---@diagnostic disable

---@class AttackTypeHitPrereqCondition : BaseHitPrereqCondition
---@field attackType gamedataAttackType
AttackTypeHitPrereqCondition = {}

---@return AttackTypeHitPrereqCondition
function AttackTypeHitPrereqCondition.new() return end

---@param props table
---@return AttackTypeHitPrereqCondition
function AttackTypeHitPrereqCondition.new(props) return end

---@param hitEvent gameeventsHitEvent
---@return Bool
function AttackTypeHitPrereqCondition:Evaluate(hitEvent) return end

---@param recordID TweakDBID|string
function AttackTypeHitPrereqCondition:SetData(recordID) return end

